bin/complete

```diff
@@ -1,26 +0,0 @@
-# The following command was used to generate this script:
-# _TMT_COMPLETE=source python3 bin/tmt > bin/complete
-# https://click.palletsprojects.com/en/7.x/bashcomplete/
-
-_tmt_completion() {
-    local IFS=$'
-'
-    COMPREPLY=( $( env COMP_WORDS="${COMP_WORDS[*]}" \
-                   COMP_CWORD=$COMP_CWORD \
-                   _TMT_COMPLETE=complete $1 ) )
-    return 0
-}
-
-_tmt_completionetup() {
-    local COMPLETION_OPTIONS=""
-    local BASH_VERSION_ARR=(${BASH_VERSION//./ })
-    # Only BASH version 4.4 and later have the nosort option.
-    if [ ${BASH_VERSION_ARR[0]} -gt 4 ] ||
-            ([ ${BASH_VERSION_ARR[0]} -eq 4 ] &&
-            [ ${BASH_VERSION_ARR[1]} -ge 4 ]); then
-        COMPLETION_OPTIONS="-o nosort"
-    fi
-    complete $COMPLETION_OPTIONS -F _tmt_completion tmt
-}
-
-_tmt_completionetup;
```

Comment:
Users still have the option to run `tmt setup completion <shell> --install`

Reply:
That's right, but it's an additional step needed. Currently, with rpm package, user just installs `tmt` and completion works out of the box, no need to study how to enable it.

    > rpm -ql tmt | grep completion
    /etc/bash_completion.d/tmt

Is it complicated to keep the file there with the new setup?

Reply:
Not complicated, just not sure about this approach. How about a [scriptlet](https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/) ?

Reply:
I don't understand. Why would a scriplet be easier than including a file in the package?

Reply:
Easier probably not. My thinking is that it is variable - Click version is not locked, users can have different bash versions, install tmt from PyPA, etc.  
Plus it's another file to maintain and while not uncommon to package it, there is no mention of it in the guidelines.
The file currently being packaged for example differs from what `tmt setup completion bash` generates now.

Reply:
> Easier probably not. My thinking is that it is variable - Click version is not locked, users can have different bash versions, install tmt from PyPA, etc.

Yes, users can install using `pip` but let's keep the user experience for `dnf` users as fluent as possible. Also, as far as I known we did not run into any bash/click version incompatibilities so far.

> Plus it's another file to maintain and while not uncommon to package it, there is no mention of it in the guidelines. The file currently being packaged for example differs from what `tmt setup completion bash` generates now.

Actually, I'd say there's not much to maintain, single commit during the last three years ;-) One step less from everybody who installs `tmt` using `dnf`.

Reply:
ok, I'll put it back. Where do you want it since `bin` directory is gone, root dir?

I wish there are some guidelines for packagers for this, as seems a bit arbitrary in my experience.

Reply:
@psss Having fun with bash-completion..

 - The completion currently shipped does not actually work. I've tried epel9, F38 and rawhide and I have no completion. Even after installing bash-completion package.

      Double-TAB makes it actually 'stuck', rather than list of current dir files, making it arguably worse then having no completion.
      Removing it and replacing it with new one makes it work:
      `tmt setup completion bash > /usr/share/bash-completion/completions/tmt` 

      > Actually, I'd say there's not much to maintain
      
      see? :)
 
 - The completion directory can be updated from legacy `/etc/bash_completion.d` to `%{_datadir}/bash-completion/completions/`, as it was only needed for epel8 as far as I can tell. Also, we should use macro instead of install.
 See: 
[bash-completion FAQ]( https://github.com/scop/bash-completion/blob/master/README.md#faq), [dnf5](https://github.com/rpm-software-management/dnf5/blob/1734e1a40affa0134f839454c38403efa6734d37/dnf5.spec#L243C37-L243C37), [podman](https://github.com/containers/podman/blob/d912e735a30ab89e3b9522338afa5f89a73a85f1/rpm/podman.spec#L363) 

- It would make sense to me to add bash-completion as `Recommends` to specfile if bash-completion file is being shipped (as seen in the dnf5 example above).